### PR TITLE
better handling of nan numbers in common APIs

### PR DESCRIPTION
**Pull request by pelikhan**
A bit more runtime defense in the case of null/undefined/NaN parameters passed to common apis.

- [x] basic.showNumber(NaN) -> ?
- [x] led.plotBarGraph(NaN) -> clear screen
- [x] pause(NaN) => pause(20)

### CODE REVIEW

File: libs/core/basic.ts [hunk 1]
```diff
@@ -10,7 +10,10 @@ namespace basic {
     //% async
     //% parts="ledmatrix" interval.defl=150
     export function showNumber(value: number, interval?: number) {
-        showString(Math.roundWithPrecision(value, 2).toString(), interval);
+        if (isNaN(value))
```
Comment (abchatra) [inline libs/core/basic.ts:13]: Shouldn't we just show NaN as a string?

File: libs/core/basic.ts [hunk 2]
```diff
@@ -19,6 +22,7 @@ namespace basic {
  * @param ms how long to pause for, eg: 100, 200, 500, 1000, 2000
  */
 function pause(ms: number): void {
+    if (isNaN(ms)) ms = 20
```
Comment (abchatra) [inline libs/core/basic.ts:25]: Good to update documentation of pause.

**Review by abchatra — CHANGES_REQUESTED** [review]
Looks good other than the showNumber change.

File: libs/core/basic.ts (hunk 1 above)
Comment (pelikhan) [reply libs/core/basic.ts:13]: NaN is meaningless for kids and in general anyone that does not have a CS background. In general, `?` felt like a more intuitive way of saying something is wrong with the input.

Also will scroll which is not great.

Comment (abchatra) [reply libs/core/basic.ts:13]: I am not sure question mark is intuitive either. Either way we have document what it means. NaN (not a number) can be used to explain? @Jaqster opinion here?

Comment (pelikhan) [reply libs/core/basic.ts:13]: I updated the docs to explain that ? is NaN... but deeper docs about NaN would be required. They've existed v1 but mostly our APIs never returned those. It's not true anymore with Jacdac when you query a sensor value where the hardware is missing.

Comment (riknoll) [reply libs/core/basic.ts:13]: I think ? is more intuitive than NaN for sure. I like this change

Comment (pelikhan) [reply libs/core/basic.ts:13]: @abchatra can we have a decision here?

Comment (pelikhan) [reply libs/core/basic.ts:13]: It's not just NaN, also "null" and "undefined" are all handled to "?".

Comment (abchatra) [reply libs/core/basic.ts:13]: I am good with your change

**Pull request merged**